### ISSUE TITLE
[BUG] All dependencies fail to deploy in GCCH

### ISSUE THREAD

**Issue by bwar01** [pnp/custom-learning-office-365#199]
Successfully deployed solution in two different tenants, however all dependencies appear to deploy in GCCH tenant.

![image](https://user-images.githubusercontent.com/13757/73865739-92e27e80-47f8-11ea-8f0f-6ff5061109d5.png)

**Comment by juliemturner** [comment]
If by GCCH you mean government cloud I'm fairly sure that you cannot run SharePoint framework solutions in the government cloud because scripting has been disabled. That said I have no access to a government cloud tenant so I don't have any way to test of validate that information.  

Assuming that answers your question but if not could you please provide more information?

**Comment by spsguru** [comment]
FYI, GCCH is GCC High government cloud tenants, I can validate that you can install and use LP in GCC (regular) tenants, but yes, scripting needs to be enabled in the tenant. There maybe an issue with the use of a CDN, as GCCH requires any CDNs or other data sources to be within the boundaries set for GCCH.

**Comment by bwar01** [comment]
> FYI, GCCH is GCC High government cloud tenants, I can validate that you can install and use LP in GCC (regular) tenants, but yes, scripting needs to be enabled in the tenant. There maybe an issue with the use of a CDN, as GCCH requires any CDNs or other data sources to be within the boundaries set for GCCH.


Interesting- I can validate the following:
GCC:

**Comment by bwar01** [comment]
Hi Julie, Apologies for the delay.  it appears that in GCC everything deploys as expected using default tenant settings (denycustompages = enabled).

In GCCH, Disabling denycustompages did not appear to have any effect.  The M365IPconfigurations.ps2 does successfully deploy the following sitepages: CustomLearningViewer.aspx and CustomLearningAdmin.aspx

Running through a test case where it appears the following libraries successfully deploy: CustomAssets, CustomConfig, CustomPlaylists but contain no data.  I'm able to copy content in those respective lists from my GCC environment and everything in GCCH appears to work.  

Navigation does not appear to successfully deploy. 
Images do not appear to deploy either- but can manually copy them over from GCC. 

Basically I'm manually restoration of dependencies.  

The LP web part also successfully deploys and works as expected. 

Re: the CDN, there are many public CDNs that MSFT services rely on.  GCCH should have no issue accessing them explicitly as needed.  

I'm not tracking how some things are able to deploy and others not?  There is no desire to leverage the provisioning engine- manually deployment will work just fine.

**Comment by juliemturner** [comment]
I can't help but comment that to have a GCCH tenant and then purposefully disable some of the security features seems counterintuitive, but that's just an opinion and certainly, you have the right to try and overcome them. 

Some of the things I know you're going to have to overcome:
- I'm fairly sure your issue deploying those pages is that the SingleAppPart page type is not supported in GCCH, you'll have to manually create a CustomLearningAdmin.aspx page and programmatically (with PowerShell) add the CustomLearningAdmin web part to it.
- To @spsguru point, I think it's unlikely that the GCCH tenant will allow you to make a call outside the tenant origin into GitHub (the CDN) to download the images and JSON files so you'll have to clone the repo and host those files somewhere else and modify the tenant app properties to point to this alternate location.
- And finally, the biggest issue is that as I understand it the GCCH tenants block iFraming in support.office.com and docs.microsoft.com content into Sharepoint pages. So even if you overcame every other hurdle I'm not sure what content you could actually see.

The long and the short of it is that this solution is not supported in GCCH tenants. Because I don't have access to one it's going to be very hard for me to help you overcome the issues with deployment. Fundamentally, you would really need to understand the architecture of the solution so that you can implement an alternative workaround for the things that will block you. I would recommend you watch this [video](https://www.youtube.com/watch?v=-sLXl_rKGxQ&t=6s) so that you can get a better understanding of the solution before you try and proceed. 

Some other things that you mentioned that I didn't address above:
- With regard to the custom lists: CustomPlaylists and CustomAssets will not have any data in them until you start trying to add your own custom data via the administration web part. CustomConfig will have a configuration list entry when the admin/viewer web part successfully loads, since they haven't that list should also be empty.
- I'm not sure I follow what you mean by Navigation, if the web parts aren't able to download content there is no navigation.

**Comment by juliemturner** [comment]
Closing issue due to inactivity.
